### ISSUE TITLE
Reporting Deprecated TLS 1.2

### ISSUE THREAD

**Issue by DhiyaneshGeek** [projectdiscovery/nuclei#3358]
### Nuclei version:
v2.8.9 (Bug in Latest Version)
v2.8.8 (Work as Intended)

### Current Behavior:
```bash
root@localhost:~# nuclei -u redacted.com -id deprecated-tls -vv

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.9

		projectdiscovery.io

[INF] Using Nuclei Engine 2.8.9 (latest)
[INF] Using Nuclei Templates 9.3.7 (latest)
[INF] Templates added in last update: 58
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 1
[deprecated-tls] Deprecated TLS Detection (TLS 1.1 or SSLv3) (@righettod,@forgedhallpass) [info]
[deprecated-tls] [ssl] [info] redacted.com [tls12]
[deprecated-tls] [ssl] [info] redacted.com [tls12]
[deprecated-tls] [ssl] [info] redacted.com [tls12]
```
### Expected Behavior:

```bash
root@localhost:~# ./nuclei -u redacted.com -id deprecated-tls -vv

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.8

		projectdiscovery.io

[INF] Using Nuclei Engine 2.8.8 (outdated)
[INF] Using Nuclei Templates 9.3.7 (latest)
[INF] Templates added in last update: 58
[INF] Templates loaded for scan: 1
[INF] Targets loaded for scan: 1
[deprecated-tls] Deprecated TLS Detection (TLS 1.1 or SSLv3) (@righettod,@forgedhallpass) [info]
[INF] No results found. Better luck next time!
```

### Anything else:
https://discord.com/channels/695645237418131507/700096242226692236/1078466426190438511

**Comment by Mzack9999** [comment]
@DhiyaneshGeek It seems already fixed on the `dev` branch:
```console
$ git rev-parse --abbrev-ref HEAD
dev
$ go run . -u redacted.com -id deprecated-tls -vv    
...
[deprecated-tls] Deprecated TLS Detection (TLS 1.1 or SSLv3) (@righettod,@forgedhallpass) [info]
[deprecated-tls] [ssl] [info] redacted.com [tls10]
[deprecated-tls] [ssl] [info] redacted.com [tls11]
``

**Comment by Mzack9999** [comment]
The problem occur because openssl is used when available, but it picks up TLS 1.2 as it ignores client global min/max TLS versions (tracked at https://github.com/projectdiscovery/tlsx/issues/203)

**Comment by Mzack9999** [comment]
Indirectly fixed by https://github.com/projectdiscovery/tlsx/issues/203
In `go.mod`:
```
...
replace github.com/projectdiscovery/tlsx => /Users/marcornvh/go/src/github.com/projectdiscovery/tlsx.
...
```
Execution:
```console
$ go run . -u redacted.com -id deprecated-tls -c 1 -bs 1
...
[deprecated-tls] [ssl] [info] redacted.com:443 [tls10]
[deprecated-tls] [ssl] [info] redacted.com:443 [tls11]
```
